modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/domain/AmbiguousDomainReferenceException.java

```diff
@@ -9,14 +9,11 @@
 import static java.lang.String.format;
 import static org.mule.runtime.api.i18n.I18nMessageFactory.createStaticMessage;
 import org.mule.runtime.api.exception.MuleException;
+import org.mule.runtime.module.artifact.api.descriptor.BundleDescriptor;
 
-/**
- * Exception type for representing a failure due to domain not found.
- */
-public final class IncompatibleDomainVersionException extends MuleException {
+public class AmbiguousDomainReferenceException extends MuleException {
 
-  public IncompatibleDomainVersionException(String expectedDomainName, String availableVersion) {
-    super(createStaticMessage(format("Expected domain '%s' couldn't be retrieved. It is available the '%s' version",
-                                     expectedDomainName, availableVersion)));
+  public AmbiguousDomainReferenceException(BundleDescriptor bundleDescriptor) {
+    super(createStaticMessage(format("More than one compatible domain were found for bundle descriptor %s", bundleDescriptor)));
```

Comment:
Would it be possible to list those ambiguous domains found?

Reply:
As part of the error message.

Reply:
I added the domain names to the error message